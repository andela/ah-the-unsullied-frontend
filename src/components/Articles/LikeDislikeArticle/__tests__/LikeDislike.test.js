import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { LikeDislike, mapStateToProps } from '../index.js';

Enzyme.configure({ adapter: new Adapter() });
describe('LikeDislike', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      likesDislikes: {},
      changelikesDislikes: {
        likes: 0,
        dislikes: 4
      },
      article: {
        article: {
          article: {
            slug: 'test'
          }
        }
      },
      likeArticle: jest.fn(slug => {
        Promise.resolve();
      }),
      dislikeArticle: jest.fn(slug => {
        Promise.resolve();
      })
    };

    state = {
      articlereducer: {
        article: {}
      },
      likesDislikes: {}
    };
    wrapper = shallow(<LikeDislike {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render the like and dislike button', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state when calling handLike', () => {
    const slug = 'test';
    wrapperInstance.handleLike();
    expect(props.likeArticle).toHaveBeenCalledWith(slug);
  });

  it('should change the state when calling handdisLike', () => {
    const slug = 'test';
    wrapperInstance.handledisLike();
    expect(props.dislikeArticle).toHaveBeenCalledWith(slug);
  });

  it('mapStateToProps', () => {
    const propsState = mapStateToProps(state);
    expect(propsState.article).toEqual(state.articlereducer);
  });

  it('Changes color When liked', () => {
    wrapper.setProps({
      likesDislikes: {
        likesDislikes: { likes: 1, dislikes: 0, liked: true, disliked: false }
      }
    });
    expect(wrapperInstance.state.likestyle).toEqual('#35A7FF');
  });

  it('Will change', () => {
    wrapper.setProps({
      likesDislikes: {
        likesDislikes: { likes: 0, dislikes: 1, liked: false, disliked: true }
      }
    });
    expect(wrapperInstance.state.dislikestyle).toEqual('#35A7FF');
  });
});
