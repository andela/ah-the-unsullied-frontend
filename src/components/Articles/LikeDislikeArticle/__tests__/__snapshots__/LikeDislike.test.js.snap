// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeDislike should render the like and dislike button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LikeDislike
    article={
      Object {
        "article": Object {
          "article": Object {
            "slug": "test",
          },
        },
      }
    }
    changelikesDislikes={
      Object {
        "dislikes": 4,
        "likes": 0,
      }
    }
    dislikeArticle={[MockFunction]}
    likeArticle={[MockFunction]}
    likesDislikes={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="white"
          onClick={[Function]}
          style={
            Object {
              "color": "",
            }
          }
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-up "
          />
        </button>,
        <button
          className="white"
          onClick={[Function]}
          style={
            Object {
              "color": "",
            }
          }
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-down"
          />
        </button>,
      ],
      "className": "likebuttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            aria-hidden="true"
            className="fa fa-thumbs-up "
          />,
          "className": "white",
          "onClick": [Function],
          "style": Object {
            "color": "",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": undefined,
            "className": "fa fa-thumbs-up ",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            aria-hidden="true"
            className="fa fa-thumbs-down"
          />,
          "className": "white",
          "onClick": [Function],
          "style": Object {
            "color": "",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": undefined,
            "className": "fa fa-thumbs-down",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="white"
            onClick={[Function]}
            style={
              Object {
                "color": "",
              }
            }
          >
            <i
              aria-hidden="true"
              className="fa fa-thumbs-up "
            />
          </button>,
          <button
            className="white"
            onClick={[Function]}
            style={
              Object {
                "color": "",
              }
            }
          >
            <i
              aria-hidden="true"
              className="fa fa-thumbs-down"
            />
          </button>,
        ],
        "className": "likebuttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              aria-hidden="true"
              className="fa fa-thumbs-up "
            />,
            "className": "white",
            "onClick": [Function],
            "style": Object {
              "color": "",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": undefined,
              "className": "fa fa-thumbs-up ",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              aria-hidden="true"
              className="fa fa-thumbs-down"
            />,
            "className": "white",
            "onClick": [Function],
            "style": Object {
              "color": "",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": undefined,
              "className": "fa fa-thumbs-down",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
